src/main/java/com/powsybl/openloadflow/dc/DcLoadFlowEngine.java

```diff
@@ -182,14 +207,41 @@ public DcLoadFlowResult run() {
         var targetVectorArray = targetVector.getArray().clone();
 
         // First linear system solution
-        boolean success = solve(targetVectorArray, context.getJacobianMatrix(), reportNode);
+        runningContext.lastSolverSuccess = solve(targetVectorArray, context.getJacobianMatrix(), reportNode);
 
         equationSystem.getStateVector().set(targetVectorArray);
         updateNetwork(network, equationSystem, targetVectorArray);
 
-        // continue with PST active power control outer loop only if first linear system solution has succeeded
-        if (success && parameters.getNetworkParameters().isPhaseControl()) {
-            success = runPhaseControlOuterLoop(phaseShifterControlOuterLoop, outerLoopContext);
+        // continue with outer loops only if solver succeed
+        if (runningContext.lastSolverSuccess) {
+            int oldSolverTotalIterations;
```

Comment:
in line with previous comment, propose to rename `oldSolverTotalIterations` to `oldSolverTotalExecutions`